### PR TITLE
CNV docs reorganization

### PR DESCRIPTION
**Pull request by apinnick**
Version(s): 4.14

Issue: None

Link to docs preview:

- [About](https://61688--docspreview.netlify.app/openshift-enterprise/latest/virt/about_virt/about-virt.html)
- [Release notes](https://61688--docspreview.netlify.app/openshift-enterprise/latest/virt/release_notes/virt-release-notes-placeholder.html) - new directory with placeholder for RN. 4.14 has its own [RN PR](https://github.com/openshift/openshift-docs/pull/61687).
- [Getting started](https://61688--docspreview.netlify.app/openshift-enterprise/latest/virt/getting_started/virt-getting-started.html)
- Installing:
    - [Preparing cluster](https://61688--docspreview.netlify.app/openshift-enterprise/latest/virt/install/preparing-cluster-for-virt.html) - moved OS/hardware requirements to assembly so we can use xrefs.
    - [Web and CLI installation assemblies](https://61688--docspreview.netlify.app/openshift-enterprise/latest/virt/install/installing-virt.html) combined
    - [Uninstalling](https://61688--docspreview.netlify.app/openshift-enterprise/latest/virt/uninstall/uninstalling-virt.html) - moved to "Install" directory
- [Updating](https://61688--docspreview.netlify.app/openshift-enterprise/latest/virt/upgrade/upgrading-virt.html)

QE review: NA
- [ ] QE has approved this change.

Additional information: Part of general CNV docs reorg. See [Miro board](https://miro.com/app/board/uXjVPu-F7y0=/) for details.

This is part 1 of the docs reorg. I am creating separate PRs to reduce the burden on the peer reviewer.

### CODE REVIEW

**Comment by cwilkers** [comment]
I like the new clean look and the way everything follows the top level OCP/OKD structure.

One thing bothers me about this structure. It's an issue with the overall way the documentation works, that if you click on a "directory", you don't get a change in content to reflect that selection. Most sections get around this issue by having at least one "page" directly under the directory, but here you have two layers of directory.

I think the Virtualization section would flow more naturally with the "About OpenShift Virtualization" left where it was, and the rest of the pages cleaned up like you're doing here.

Unfortunately, that leaves the problem of what to do with the other pages in the "About" section, whether to move those pages or rename the directory. I'm struggling to come up with a better layout, honestly.

**Comment by dominikholler** [comment]
Looks like a nice improvement.

**Comment by apinnick** [comment]
> One thing bothers me about this structure. It's an issue with the overall way the documentation works, that if you click on a "directory", you don't get a change in content to reflect that selection. Most sections get around this issue by having at least one "page" directly under the directory, but here you have two layers of directory.

@cwilkers There are technical limitations, since the HTML is generated by Asciibinder. Directories are actually folders, so they cannot "change" content. They only display their contents. However, there are some workarounds that I can try.

File: virt/install/preparing-cluster-for-virt.adoc [hunk 1]
```diff
@@ -19,49 +26,78 @@ You can use any installation method, including user-provisioned, installer-provi
...
+* On-premise bare metal servers. See xref:../../installing/installing_bare_metal/preparing-to-install-on-bare-metal.adoc#virt-planning-bare-metal-cluster-for-ocp-virt_preparing-to-install-on-bare-metal[Planning a bare metal cluster for {VirtProductName}].
+* Amazon Web Services bare metal instances. See link:https://access.redhat.com/articles/6409731[Deploy OpenShift Virtualization on AWS metal instance types].
 
-[role="_additional-resources"]
-.Additional resources
+ifdef::openshift-enterprise[]
+:FeatureName: Installing OpenShift Virtualization on AWS bare metal instances
+include::snippets/technology-preview.adoc[]
+:!FeatureName:
+endif::[]
```
Comment (ousleyp) [inline virt/install/preparing-cluster-for-virt.adoc:45]: For the bullet points with TP blocks, I would nest the TP include under the relevant bullet point. For example:

```
* Amazon Web Services bare metal instances. See link:https://access.redhat.com/articles/6409731[Deploy OpenShift Virtualization on AWS metal instance types].
+
--
ifdef::openshift-enterprise[]
:FeatureName: Installing OpenShift Virtualization on AWS bare metal instances
include::snippets/technology-preview.adoc[]
:!FeatureName:
endif::[]
--

* IBM Cloud Bare Metal Servers. See link:https://access.redhat.com/articles/6738731[Deploy OpenShift Virtualization on IBM Cloud Bare Metal nodes].
+
--
ifdef::openshift-enterprise[]
:FeatureName: Installing OpenShift Virtualization on IBM Cloud Bare Metal Servers
include::snippets/technology-preview.adoc[]
:!FeatureName:
endif::[]
--
```

File: virt/install/virt-specifying-nodes-for-virtualization-components.adoc [hunk 2]
```diff
@@ -1,7 +1,7 @@
 :_content-type: ASSEMBLY
 include::_attributes/common-attributes.adoc[]
 [id="virt-specifying-nodes-for-virtualization-components"]
-= Specifying nodes for {VirtProductName} components
+= Specifying nodes for {VirtProductName} deployment
```
Comment (ousleyp) [inline virt/install/virt-specifying-nodes-for-virtualization-components.adoc:4]: I'm curious where this change originated - is it necessary?

File: virt/post_install/index.adoc [hunk 3]
```diff
@@ -0,0 +1,91 @@
+:_content-type: ASSEMBLY
+[id="post-install-configuration-overview"]
+= Post-installation configuration overview
+include::_attributes/common-attributes.adoc[]
+:context: post-installation-configuration-overview
+
+toc::[]
+
+COPIED FROM OCP post-install
```
Comment (ousleyp) [inline virt/post_install/index.adoc:9]: This should probably be commented out even if this is a WIP/not published.

File: virt/post_install/index.adoc [hunk 4]
```diff
@@ -0,0 +1,91 @@
+:_content-type: ASSEMBLY
+[id="post-install-configuration-overview"]
+= Post-installation configuration overview
```
Comment (ousleyp) [inline virt/post_install/index.adoc:3]: I fear that this will be hard to maintain. Maybe create a symbolic link to the assembly if you're planning to keep it the same? Or is this meant to be a template for creating a similar page of our own? (I see that it isn't in the topic map yet, so perhaps that's your plan.)

I'd request that you make this addition in its own PR, either way.

File: virt/release_notes/virt-release-notes-placeholder.adoc [hunk 5]
```diff
@@ -1,12 +1,11 @@
...
+[id="virt-release-notes-placeholder"]
+= {VirtProductName} release notes placeholder
 include::_attributes/common-attributes.adoc[]
-:context: virt-release-notes
-
+:context: virt-release-notes-placeholder
 toc::[]
 
 Do not add or edit release notes here. Edit release notes directly in the branch
 that they are relevant for.
```
Comment (ousleyp) [inline virt/release_notes/virt-release-notes-placeholder.adoc:9]: As mentioned elsewhere, this change applies to main only. Please open a separate PR for it. Thanks!

File: virt/upgrade/_attributes [hunk 6]
```diff
@@ -0,0 +1 @@
+../../_attributes/
```
Comment (ousleyp) [inline virt/upgrade/_attributes:1]: Not sure if you will need to re-do anything, link-wise, if you change the dir name to "updating." Maybe not?

**Review by ousleyp — COMMENTED** [review]
Hey Avital, I love how impactful the reorganization is, especially when you look at the ToC. It's much more pleasant to look at. I do have a number of comments for you to consider, including a couple requests for you to split things out into separate PRs. Please let me know if you have any questions about my comments; I'm happy to talk it out. :) Thanks for doing this work!

File: _topic_maps/_topic_map.yml [hunk 7]
```diff
@@ -3594,72 +3594,79 @@ Name: Virtualization
...
+  - Name: Security policies
+    File: virt-security-policies
+  - Name: Architecture
+    File: virt-architecture
+    Distros: openshift-enterprise
+- Name: Release notes
+  Dir: release_notes
+  Topics:
+  - Name: OpenShift Virtualization release notes
+    File: virt-release-notes-placeholder
```
[the remark this answers is not in the document]
Comment (apinnick) [reply _topic_maps/_topic_map.yml:3615]: Good idea. The RN changes are a real headache. I will create a separate PR.

File: virt/post_install/index.adoc (hunk 3 above)
Comment (apinnick) [reply virt/post_install/index.adoc:9]: Oops. I meant to put that in a separate PR. Will delete!

File: virt/about_virt/virt-architecture.adoc [hunk 8]
```diff
@@ -20,3 +21,25 @@ After all operator pods are successfully deployed, you should create the `HyperC
 The `HyperConverged` CR creates corresponding CRs for the operators of all other components within its reconciliation loop. Each operator then creates resources such as daemon sets, config maps, and additional components for the {VirtProductName} control plane. For example, when the `hco-operator` creates the `KubeVirt` CR, the `virt-operator` reconciles it and create additional resources such as `virt-controller`, `virt-handler`, and `virt-api`.
 
 The OLM deploys the `hostpath-provisioner-operator`, but it is not functional until you create a `hostpath provisioner` (HPP) CR.
+
+image::cnv_components_main.png[Deployments]
+
+[role="_additional-resources"]
+.Additional resources
+* link:https://github.com/kubevirt/hyperconverged-cluster-operator/blob/main/docs/cluster-configuration.md[HyperConverged CR configuration]
```
[the remark this answers is not in the document]
Comment (apinnick) [reply virt/about_virt/virt-architecture.adoc:29]: Good catch! I will look for something downstream.

File: virt/install/preparing-cluster-for-virt.adoc [hunk 9]
```diff
@@ -10,7 +10,14 @@ Review this section before you install {VirtProductName} to ensure that your clu
...
 ====
-You can use any installation method, including user-provisioned, installer-provisioned, or assisted installer, to deploy {product-title}. However, the installation method and the cluster topology might affect {VirtProductName} functionality, such as snapshots or live migration.
+Installation method considerations::
+You can use any installation method, including user-provisioned, installer-provisioned, or assisted installer, to deploy {product-title}. However, the installation method and the cluster topology might affect {VirtProductName} functionality, such as snapshots or xref:../../virt/install/preparing-cluster-for-virt.adoc#live-migration_preparing-cluster-for-virt[live migration].
+
+{rh-storage-first}::
+If you deploy {VirtProductName} with {rh-storage-first}, you must create a dedicated storage class for Windows virtual machine disks. See link:https://access.redhat.com/articles/6978371[Optimizing ODF PersistentVolumes for Windows VMs] for details.
+
+IPv6::
+You cannot run {VirtProductName} on a single-stack IPv6 cluster. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2193267[*BZ#2193267*])
```
[the remark this answers is not in the document]
Comment (apinnick) [reply virt/install/preparing-cluster-for-virt.adoc:20]: Good point. Will remove the link. I thought it was a KI, but I was mistaken.

File: virt/install/preparing-cluster-for-virt.adoc (hunk 1 above)
Comment (apinnick) [reply virt/install/preparing-cluster-for-virt.adoc:45]: Doesn't work. I tried that.

File: modules/virt-about-storage-volumes-for-vm-disks.adoc [hunk 10]
```diff
@@ -1,27 +1,27 @@
...
+If you use the storage API with known storage providers, the volume and access modes are selected automatically. However, if you use a storage class that does not have a storage profile, you must configure the volume and access mode.
 
-For best results, use `accessMode: ReadWriteMany` and `volumeMode: Block`. This is important for the following reasons:
+For best results, use the `ReadWriteMany` (RWX) access mode and the `Block` volume mode. This is important for the following reasons:
 
-* The ReadWriteMany (RWX) access mode is required for live migration.
+* `ReadWriteMany` (RWX) access mode is required for live migration.
 
-* The `Block` volume mode performs significantly better in comparison to the `Filesystem` volume mode. This is because the `Filesystem` volume mode uses more storage layers, including a file system layer and a disk image file. These layers are not necessary for VM disk storage.
+* `Block` volume mode performance is significantly better than `Filesystem`. This is because the `Filesystem` volume mode uses more storage layers, including a file system layer and a disk image file. These layers are not necessary for VM disk storage.
```
[the remark this answers is not in the document]
Comment (apinnick) [reply modules/virt-about-storage-volumes-for-vm-disks.adoc:16]: I'm not sure what you mean. Should I revert some or all my changes?

File: virt/install/preparing-cluster-for-virt.adoc (hunk 1 above)
Comment (apinnick) [reply virt/install/preparing-cluster-for-virt.adoc:46]: Very cool! I will give that a try.

File: modules/virt-installing-virt-operator.adoc [hunk 11]
```diff
@@ -1,12 +1,12 @@
 // Module included in the following assemblies:
 //
-// * virt/install/installing-virt-web.adoc
+// * virt/install/installing-virt.adoc
 
 :_content-type: PROCEDURE
 [id="virt-installing-virt-operator_{context}"]
```
[the remark this answers is not in the document]
Comment (apinnick) [reply modules/virt-installing-virt-operator.adoc:6]: I will revert the title.

File: modules/virt-enabling-virt-repos.adoc [hunk 12]
```diff
@@ -1,6 +1,6 @@
 // Module included in the following assemblies:
 //
-// * virt/install/virt-installing-virtctl.adoc
+// * virt/getting_started/virt-using-the-cli-tools.adoc
```
[the remark this answers is not in the document]
Comment (apinnick) [reply modules/virt-enabling-virt-repos.adoc:3]: Good catch. Will delete.

File: virt/install/installing-virt.adoc [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+[id="installing-restricted-network_installing-virt"]
+== Installing in a restricted network environment
+
+If you install {VirtProductName} in a restricted environment with no internet connectivity, you must xref:../../operators/admin/olm-restricted-networks.adoc#olm-restricted-networks[configure Operator Lifecycle Manager (OLM) for restricted networks].
+
+If you have limited internet connectivity, you can xref:../../operators/admin/olm-configuring-proxy-support.adoc#olm-configuring-proxy-support[configure proxy support in OLM] to access the OperatorHub.
+
+[id="deploying-virt-operator_installing-virt"]
+== Deploying the {VirtProductName} Operator
+
```
[the remark this answers is not in the document]
Comment (apinnick) [reply virt/install/installing-virt.adoc:24]: I know you prefer the direct approach :-) - "Do this by using X or Y". I am still using "You can do this by using X or Y." Haven't yet made up my mind but I will add th sentence and change 'deploying' to 'installing'.

File: virt/install/preparing-cluster-for-virt.adoc [hunk 14]
```diff
@@ -19,49 +26,78 @@ You can use any installation method, including user-provisioned, installer-provi
...
+endif::[]
 
-* xref:../../architecture/architecture-rhcos.adoc#rhcos-about_architecture-rhcos[About RHCOS].
-* link:https://catalog.redhat.com[Red Hat Ecosystem Catalog] for supported CPUs.
-* xref:../../storage/index.adoc#storage-overview[Supported storage].
+* IBM Cloud Bare Metal Servers. See link:https://access.redhat.com/articles/6738731[Deploy OpenShift Virtualization on IBM Cloud Bare Metal nodes].
 
-include::modules/virt-cluster-resource-requirements.adoc[leveloffset=+1]
+ifdef::openshift-enterprise[]
+:FeatureName: Installing OpenShift Virtualization on IBM Cloud Bare Metal Servers
```
[the remark this answers is not in the document]
Comment (apinnick) [reply virt/install/preparing-cluster-for-virt.adoc]: Will change

File: modules/virt-about-storage-volumes-for-vm-disks.adoc (hunk 10 above)
[the remark this answers is not in the document]
Comment (apinnick) [reply modules/virt-about-storage-volumes-for-vm-disks.adoc:16]: I reverted the first sentence.

**Comment by peterclauterbach** [comment]
Overall, this looks god. However, the VM networking and VM storage sections are still buried in "virtual machines", which makes that section very long and complex.
Are we considering pulling those two sections up on level, so they are easier to find?
